File: JavaScript/7js 4/sumOfMinimums.js
/*
|--------------------------------------------------------------------------
| Are the numbers in order?
|--------------------------------------------------------------------------
|
*/

function sumOfMinimums(arr) {
   let res = []
   for (let i = 0; i < arr.length; i++) {
      res.push(Math.min(...arr[i]))
   }
   return res.reduce((a, b) => a + b);
}


const sumOfMinimums = (array) => array.map(el => Math.min(...el)).reduce((a, b) => a + b, 0)


const sumOfMinimums = (arr) => arr.reduce((a, b) => a + b.sort((x, z) => x - z)[0], 0)


function sumOfMinimums(arr) {
   return arr.reduce((p, c) => p + Math.min(...c), 0);
}


const _ = require('lodash/fp');
const sumOfMinimums = _.sumBy(_.min);


console.log(sumOfMinimums([
   [7, 9, 8, 6, 2],
   [6, 3, 5, 4, 3],
   [5, 8, 7, 4, 5]
]), 9);
console.log(sumOfMinimums([
   [11, 12, 14, 54],
   [67, 89, 90, 56],
   [7, 9, 4, 3],
   [9, 8, 6, 7]
]), 76);

File: Python/7py 3/sum_of_minimums.py
"""
|--------------------------------------------------------------------------
| Are the numbers in order?
|--------------------------------------------------------------------------
|
"""

def sum_of_minimums(numbers):
   res = []
   for i in range(len(numbers)):
      res.append(min(numbers[i]))
   return sum(res)

def sum_of_minimums(numbers):
   return sum([min(x) for x in numbers])


def sum_of_minimums(numbers):
   return sum(map(min, numbers))


def sum_of_minimums(numbers):
   return sum([sorted(x)[0] for x in numbers])
   


print(sum_of_minimums([ [7,9,8,6,2 ], [6,3,5,4,3], [5,8,7,4,5] ]),9)
print(sum_of_minimums([ [11,12,14,54], [67,89,90,56], [7,9,4,3], [9,8,6,7]]),76)